bindings/java/src/test/com/apple/foundationdb/test/StackTester.java

```diff
@@ -547,18 +547,15 @@ void processOp(byte[] operation) {
 
 		@Override
 		void executeOperations() {
-			KeySelector begin = nextKey;
 			while(true) {
-				Transaction t = db.createTransaction();
-				List<KeyValue> keyValues = t.getRange(begin, endKey/*, 1000*/).asList().join();
-				t.close();
+				List<KeyValue> keyValues = db.read(readTr -> readTr.getRange(nextKey, endKey/*, 1000*/).asList().join());
 				if(keyValues.size() == 0) {
 					break;
 				}
 				//System.out.println(" * Got " + keyValues.size() + " instructions");
 
 				for(KeyValue next : keyValues) {
-					begin = KeySelector.firstGreaterThan(next.getKey());
+					nextKey = KeySelector.firstGreaterThan(next.getKey());
```

Comment:
Since `nextKey` wasn't updated in earlier code, I presume it was a bug that could have allowed running `executeOperations()` twice?

Reply:
`executeOperations` is only ever called once, and `nextKey` was only being used at the previous line 550 to assign to `begin`. This change gets rid of the unnecessary intermediary `begin` and makes the logic match `AsyncStackTester`. I'm not really sure why it was done this way here, but my guess is that it probably happened accidentally with some sort of refactoring.